ui/app/AppLayouts/Chat/panels/communities/TokenHoldersPanel.qml

```diff
@@ -64,26 +61,87 @@ Control {
-            leftMargin: -Style.current.padding
-            model: filteredModel
-            delegate: StatusListItem {
-                readonly property bool unknownHolder: model.ensName === ""
-                readonly property string formattedTitle: unknownHolder ? "?" : model.ensName
 
-                sensor.enabled: false
+            model: filteredModel
+            delegate: RowLayout {
                 width: ListView.view.width
-                title: formattedTitle
-                statusListItemTitle.visible: !unknownHolder
-                subTitle: model.walletAddress
-                asset.name: model.imageSource
-                asset.isImage: true
-                asset.isLetterIdenticon: unknownHolder
-                asset.color: Theme.palette.userCustomizationColors[d.red2Color]
+                spacing: Style.current.padding
+
+                StatusListItem {
+                    readonly property bool unknownHolder: model.ensName === ""
+                    readonly property string formattedTitle: unknownHolder ? "?" : model.ensName
+
+                    Layout.fillWidth: true
+
+                    leftPadding: 0
+                    rightPadding: 0
+                    sensor.enabled: false
+                    title: formattedTitle
+                    statusListItemTitle.visible: !unknownHolder
+                    subTitle: model.walletAddress
+                    asset.name: model.imageSource
+                    asset.isImage: true
+                    asset.isLetterIdenticon: unknownHolder
+                    asset.color: Theme.palette.userCustomizationColors[d.red2Color]
+                }
+
+                StatusComboBox {
+                    id: combo
+
+                    Layout.preferredWidth: 70
+                    Layout.preferredHeight: 44
+
+                    visible: root.isSelectorMode && amount > 1
+                    control.spacing: Style.current.halfPadding / 2
+                    model: amount
+                    size: StatusComboBox.Size.Small
+                    type: StatusComboBox.Type.Secondary
+                    delegate: StatusItemDelegate {
+                        width: combo.control.width
+                        textHorizontalAligment: Text.AlignHCenter
+                        highlighted: combo.control.highlightedIndex === index
+                        font: combo.control.font
+                        text: Number(modelData) + 1
+                    }
+                    contentItem: StatusBaseText {
+                        font: combo.control.font
+                        verticalAlignment: Text.AlignVCenter
+                        elide: Text.ElideRight
+                        text: Number(combo.control.displayText) + 1
+                        color: Theme.palette.baseColor1
+                    }
+
+                    control.onDisplayTextChanged: {
+                        selfDestructAmount = Number(combo.control.displayText) + 1
```

Comment:
This should be sth like `combo.currentValue`; ie it should not rely on the text displayed but rather on the internal value from the model